api/transport/response.go

```diff
@@ -68,3 +68,64 @@ type ResponseWriter interface {
 	// of Write().
 	SetApplicationError()
 }
+
+// ResponseMetaWriter returns a ResponseMeta struct that handlers and inbound
+// middleware may use to modify transport.Response fields.
+//
+// Middleware and handlers may attempt to upcast ResponseWriters to
+// ResponseMetaWriters to access and write response metadata. Failure to cast
+// MUST be handled.
+//
+//  if metaW, ok := resW.(transport.ResponseMetaWriter); ok {
+//   if meta := metaW.ResponseMeta(); meta != nil{
+//     meta.Host = "foo"
+//     ...
+//   }
+//  }
+//
+// Transport implementations that support writing response metadata should have
+// their ResponseWriters implement ResponseMetaWriter to facilitate this.
+type ResponseMetaWriter interface {
+	ResponseMeta() *ResponseMeta
```

Comment:
Neither really take precedence since you could for eg:
1. write using the original `ResponseWriter`
2. write using response meta
3. write using the original `ResponseWriter`

After transports are updated, the given `ResponseWriter` will simply write to the underlying `ResponseMeta`. So although strange, users *could* use both. 

I'll update the comment to reflect that you *should* use the `ResponseMetaWriter` on successful casts, and that the `ResponseMeta` changes can affect previous `ResponseWriter` calls.